File: P5/BigNode.java
package kcore;

public class BigNode {
	private int number;
	private int c;
	private int eta;
	private boolean delete;
	//private int degree;
	
	public boolean isDelete() {
		return delete;
	}

	public void setDelete(boolean delete) {
		this.delete = delete;
	}

	public BigNode() {
		this.number = 0;
		this.c = 0;
		this.eta = 0;
		//this.degree = 0;
	}
	
	public BigNode(int number, int c, int eta) {
		this.number = number;
		this.c = c;
		this.eta = eta;
		//this.degree = degree;
	}

	public int getNumber() {
		return number;
	}

	public void setNumber(int number) {
		this.number = number;
	}

	public int getC() {
		return c;
	}

	public void setC(int c) {
		this.c = c;
	}

	public int getEta() {
		return eta;
	}

	public void setEta(int eta) {
		this.eta = eta;
	}

	@Override
	public String toString() {
		return "BigNode [node=" + number + ", c=" + c + ", eta=" + eta + "]";
	}

	/*public int getDegree() {
		return degree;
	}

	public void setDegree(int degree) {
		this.degree = degree;
	}*/

	/*@Override
	public String toString() {
		return "BigNode [node=" + node + ", c=" + c + ", eta=" + eta + ", degree=" + degree + "]";
	}*/
	
	
}

File: P5/BigNode.cpp
/*
 * BigNode.cpp
 *
 *  Created on: 21 oct. 2017
 *      Author: Charles
 */

#include "BigNode.hpp"

using namespace std;

BigNode::BigNode(){
	number = 0;
	c = 0;
	eta = 0;
}

BigNode::BigNode(unsigned long int n)
{
	number = n;
	c = 0;
	eta = 0;
	//del = false;
}


unsigned long int BigNode::getC() {
	return c;
}

void BigNode::setC(unsigned long int c) {
	this->c = c;
}

unsigned long int BigNode::getEta() {
	return eta;
}

void BigNode::setEta(unsigned long int eta) {
	this->eta = eta;
}

/*bool BigNode::isDel() {
	return del;
}

void BigNode::setIsDel(bool del) {
	this->del = del;
}*/

int BigNode::getNumber() {
	return number;
}

void BigNode::setNumber(unsigned long int number) {
	this->number = number;
}
